values/values.go

```diff
@@ -17,6 +17,15 @@ type Typer interface {
 	Type() semantic.MonoType
 }
 
+// ITableObject serves as sort of a "marker trait" to allow us to check if a
+// value is a TableObject without having to import TableObject which would be a
+// cyclical import.
+// Identical purpose to the interface in the interpreter package, but sadly we
+// can't import it here because of yet another potential cycle.
+type ITableObject interface {
```

Comment:
Can we name this `TableObject`? I'm not a big fan of the `I` prefix for interfaces and that's not really used in Go.

Is it also possible to have the method be unexported? I don't know if that's possible. If it's not, this is good. If it is possible, I'd prefer this method to be unexported.

Reply:
Happy to rename it. I was _sort of_ concerned that it'd be confusing to reuse the struct's name here, but I concede that the name I picked doesn't help clarify much beyond "this is different than `TableObject`".

I think I started out with the method being private but the implementation over on the struct didn't seem to satisfy the interface in that case. I'll try it again just to be sure.

Reply:
> I think I started out with the method being private but the implementation over on the struct didn't seem to satisfy the interface in that case. I'll try it again just to be sure.

Ah I suspected that might be the case. If that is, then it's fine if the method is exported.